### ISSUE TITLE
Rule proposal: export Declaration Merging are allowed in a namespace

### ISSUE THREAD

**Issue by ttttttssssss** [typescript-eslint/typescript-eslint#4248]
### Description

lt should be an extension rules . Because [I am using a rule from ESLint core, and it doesn't work correctly with TypeScript code](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/TROUBLESHOOTING.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code)

### Fail

```ts

namespace P{
  // error: Multiple exports of name 'f'. eslint(import/export)
  export function f () {
  }
  // error: Multiple exports of name 'f'. eslint(import/export)
  export namespace f {}
}


```

### Pass

```ts

namespace P{
  export function f () {
  }

  export namespace f {}
}


```

**Comment by ttttttssssss** [comment]
The extension-rules web page(https://github.com/typescript-eslint/typescript-eslint/blob/packages/eslint-plugin/README.md#extension-rules)  is 404

**Comment by armano2** [comment]
@ttttttssssss  correct, in this url youa re missing branch name, eg. main

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/README.md#extension-rules

**Comment by bradzacher** [comment]
> lt should be an extension rules . Because I am using a rule from ESLint core, and it doesn't work correctly with TypeScript code

What should be?
Could you provide some information?
For example... The rule name?

**Comment by ttttttssssss** [comment]
@bradzacher 

First, the reason for the error comes from [import/export](https://github.com/import-js/eslint-plugin-import/blob/v2.25.2/docs/rules/export.md) rule. 
> eslint tip : Multiple exports of name 'f'. eslint(import/export)

My advice would be to modify the core rules`@typescript-eslint/recommended` . Because it does not need to be configured for future use.

Or add a new [extension rule](https://typescript-eslint.io/rules/#extension-rules). The rule can be called `@typescript-eslint/namespace-export-declaration-merging`.

> reference: [declaration-merging document](https://www.typescriptlang.org/docs/handbook/declaration-merging.html)

**Comment by bradzacher** [comment]
`eslint-plugin-import`, as it is not eslint core, supports TS syntax.
Thus it would be much better raising an issue with them so that they can support this usecase.

We do not create extension rules for rules from other plugins.

You're also not specifically asking for "export from a namepsace" - you're just asking for declaration merged.
